applications/asset_tracker_v2/src/main.c

```diff
@@ -388,7 +388,8 @@ static void data_get(void)
 		app_module_event->data_list[count++] = APP_DATA_MODEM_STATIC;
 	}
 
-	if (!app_cfg.no_data.neighbor_cell || !app_cfg.no_data.gnss || !app_cfg.no_data.wifi) {
+	if (IS_ENABLED(CONFIG_LOCATION_MODULE) &&
```

Comment:
@gregersrygg Do we have similar issue with `APP_DATA_ENVIRONMENTAL` and `CONFIG_SENSOR_MODULE`?

Reply:
Yes, if Sensor module is disabled, battery and environmental data should not be sampled. I added flagging for these as well.